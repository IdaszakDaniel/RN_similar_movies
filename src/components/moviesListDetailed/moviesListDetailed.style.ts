import { StyleSheet, ViewStyle, ImageStyle, TextStyle } from 'react-native'
import { responsiveWidth } from '../../utils/dimensions'

interface Style {
  listContainer: ViewStyle
  itemContainer: ViewStyle
  image: ImageStyle
  imageBackground: ViewStyle
  imageStyle: ImageStyle
  title: TextStyle
  shadowView: ViewStyle
}

const styles = StyleSheet.create<Style>({
  listContainer: {
    overflow: 'hidden'
  },
  itemContainer: {
    width: responsiveWidth(95),
    height: 250,
    marginHorizontal: responsiveWidth(2),
    marginVertical: 5,
    overflow: 'hidden'
  },
  image: {
    width: 100,
    height: 100
  },
  imageBackground: {
    width: '100%',
    height: '100%'
  },
  imageStyle: {
    borderRadius: 10,
    resizeMode: 'cover'
  },
  title: {
    position: 'absolute',
    bottom: 10,
    left: 10,
    color: 'white',
    fontWeight: 'bold',
    fontSize: 24
  },
  shadowView: {
    width: '100%',
    height: '100%',
    borderRadius: 10,
    backgroundColor: 'rgba(0,0,0,0.5)',
    padding: 10
  }
})

export default styles
