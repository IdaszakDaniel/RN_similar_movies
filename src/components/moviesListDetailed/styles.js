import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  itemContainer: {
    marginHorizontal: 20,
    marginVertical: 5,
    borderBottomColor: 'gray',
    borderBottomWidth: 1
  },
  image: {
    width: 100,
    height: 100
  }
})

export default styles
