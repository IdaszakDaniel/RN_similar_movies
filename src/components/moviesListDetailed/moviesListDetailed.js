import React from 'react'
import { Text, FlatList, View, ImageBackground } from 'react-native'
import styles from './styles'
import { tmdbImage } from '../../utils/images'

const MoviesListDetailed = ({ moviesList }) => {
  const keyExtractor = item => item.id
  const renderItem = ({ item }) => {
    const uri = tmdbImage(item.data.poster_path)
    const { itemContainer, imageBackground, imageStyle, title, shadowView } = styles
    return (
      <View style={itemContainer}>
        <ImageBackground source={{ uri }} style={imageBackground} imageStyle={imageStyle}>
          <View style={shadowView}>
            <Text style={title}>{item.title}</Text>
          </View>
        </ImageBackground>
      </View>
    )
  }

  return (
    <FlatList
      data={moviesList}
      renderItem={renderItem}
      keyExtractor={keyExtractor}
      style={styles.listContainer}
    />
  )
}

export default MoviesListDetailed
