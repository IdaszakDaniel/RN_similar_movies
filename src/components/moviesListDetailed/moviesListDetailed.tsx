/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react'
import {
  Text,
  FlatList,
  View,
  ImageBackground,
  TouchableOpacity,
} from 'react-native'
import { withNavigation, NavigationInjectedProps } from 'react-navigation'
import styles from './moviesListDetailed.style'
import { tmdbImage } from '../../utils/images'

type keyType = {
  data: {
    poster_path: string
    id: number
  }
  title: string
}

type itemType = {
  item: keyType
}

export interface Props extends NavigationInjectedProps {
  movies: ReadonlyArray<keyType>
}

export const MoviesListDetailed: React.FC<Props> = ({
  movies,
  navigation: { navigate },
}): JSX.Element => {
  const keyExtractor = (item: keyType): string => item.data.id.toString()

  const renderItem = ({ item }: itemType): React.ReactElement => {
    const uri = tmdbImage(item.data.poster_path)
    const {
      itemContainer,
      imageBackground,
      imageStyle,
      title,
      shadowView,
    } = styles

    return (
      <TouchableOpacity
        onPress={(): Boolean => navigate('Details')}
        testID="item">
        <View style={itemContainer}>
          <ImageBackground
            source={{ uri }}
            style={imageBackground}
            imageStyle={imageStyle}>
            <View style={shadowView}>
              <Text style={title}>{item.title}</Text>
            </View>
          </ImageBackground>
        </View>
      </TouchableOpacity>
    )
  }

  return (
    <FlatList
      testID="flatlist"
      data={movies}
      renderItem={renderItem}
      keyExtractor={keyExtractor}
      style={styles.listContainer}
    />
  )
}

export default withNavigation(MoviesListDetailed)
