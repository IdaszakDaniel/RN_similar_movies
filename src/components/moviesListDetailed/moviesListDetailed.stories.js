import { action } from '@storybook/addon-actions'
import { text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react-native'
import React from 'react'
import CenterView from '../../../storybook/stories/CenterView'
import { MoviesListDetailed } from './moviesListDetailed'

const movies = [
  { data: { poster_path: 'poster1.png', id: 1 }, title: 'testTitle1' },
  { data: { poster_path: 'poster2.png', id: 2 }, title: 'testTitle2' }
]

const props = {
  movies,
  navigation: () => {}
}

storiesOf('Movies List Detailed', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('with two empty posters', () => <MoviesListDetailed {...props} />)
