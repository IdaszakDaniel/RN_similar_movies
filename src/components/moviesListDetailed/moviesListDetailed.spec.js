import React from 'react'
import { render, debug } from 'react-native-testing-library'
import { MoviesListDetailed } from './moviesListDetailed'

const movies = [
  { data:{ poster_path: 'poster1.png', id: 1 }, title: 'testTitle1' },
  { data:{ poster_path: 'poster2.png', id: 2 }, title: 'testTitle2' }
]

const props = {
  movies,
  navigation: () => {}
}

describe('MoviesListDetailed', () => {
  describe('1...', () => {
    it('renders properly', () => {
      const tree = render(<MoviesListDetailed {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})


