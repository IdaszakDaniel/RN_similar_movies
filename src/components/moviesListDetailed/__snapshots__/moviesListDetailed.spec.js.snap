// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesListDetailed 1... renders properly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Object {
          "id": 1,
          "poster_path": "poster1.png",
        },
        "title": "testTitle1",
      },
      Object {
        "data": Object {
          "id": 2,
          "poster_path": "poster2.png",
        },
        "title": "testTitle2",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "overflow": "hidden",
    }
  }
  testID="flatlist"
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="item"
      >
        <View
          style={
            Object {
              "height": 250,
              "marginHorizontal": 15,
              "marginVertical": 5,
              "overflow": "hidden",
              "width": 712.5,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://image.tmdb.org/t/p/w185/poster1.png",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "borderRadius": 10,
                    "resizeMode": "cover",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0.5)",
                  "borderRadius": 10,
                  "height": "100%",
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "bottom": 10,
                    "color": "white",
                    "fontSize": 24,
                    "fontWeight": "bold",
                    "left": 10,
                    "position": "absolute",
                  }
                }
              >
                testTitle1
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="item"
      >
        <View
          style={
            Object {
              "height": 250,
              "marginHorizontal": 15,
              "marginVertical": 5,
              "overflow": "hidden",
              "width": 712.5,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "http://image.tmdb.org/t/p/w185/poster2.png",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  Object {
                    "borderRadius": 10,
                    "resizeMode": "cover",
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0.5)",
                  "borderRadius": 10,
                  "height": "100%",
                  "padding": 10,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "bottom": 10,
                    "color": "white",
                    "fontSize": 24,
                    "fontWeight": "bold",
                    "left": 10,
                    "position": "absolute",
                  }
                }
              >
                testTitle2
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
